inbm/dispatcher-agent/tests/unit/source/test_ubuntu_source_cmd.py

```diff
@@ -192,14 +193,44 @@ def test_update_sources_os_error(self):
 
 
 class TestUbuntuApplicationSourceManager:
-    @patch("dispatcher.source.ubuntu_source_manager.move_file")
-    def test_update_app_source_successfully(self, mock_move):
+    def test_add_app_with_gpg_key_successfully(self):
+        try:
+            params = ApplicationAddSourceParameters(
+                file_name="intel-gpu-jammy.list",
+                sources="deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main",
+                gpg_key_uri="https://dl-ssl.google.com/linux/linux_signing_key.pub",
+                gpg_key_name="google-chrome.gpg"
+            )
+            command = UbuntuApplicationSourceManager()
+            with patch("builtins.open", new_callable=mock_open()):
+                command.add(params)
+        except SourceError as err:
+            assert False, f"'UbuntuApplicationSourceManager.add' raised an exception {err}"
+
+    def test_add_app_deb_822_format_successfully(self):
+        try:
+            params = ApplicationAddSourceParameters(
+                file_name="google-chrome.sources",
+                sources="X-Repolib-Name: Google Chrome"
+                        "Enabled: yes"
+                        "Types: deb"
+                        "URIs: https://dl-ssl.google.com/linux/linux_signing_key.pub"
+                        "Suites: stable"
+                        "Components: main",
```

Comment:
**Suggestion:** The string concatenation in the 'sources' field of 'ApplicationAddSourceParameters' seems to be missing spaces between the concatenated strings. This will result in an incorrect single string where the parts are not separated by spaces, which could lead to errors when parsing the sources. [bug]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
            params = ApplicationAddSourceParameters(
                file_name="google-chrome.sources",
                sources="X-Repolib-Name: Google Chrome "
                        "Enabled: yes "
                        "Types: deb "
                        "URIs: https://dl-ssl.google.com/linux/linux_signing_key.pub "
                        "Suites: stable "
                        "Components: main",
            )
```